src/components/DonationItemsTable/DonationsModal.jsx

```diff
@@ -0,0 +1,199 @@
+import { useEffect, useState } from 'react';
+import { useBackend } from '../../contexts/BackendContext';
+import PropTypes from 'prop-types';
+import {
+  Input,
+  Text,
+  Modal,
+  ModalContent,
+  ModalHeader,
+  ModalBody,
+  ModalOverlay,
+  ModalFooter,
+  IconButton,
+  Button,
+  Select,
+  Alert,
+  AlertTitle,
+  AlertDescription,
+  useToast,
+} from '@chakra-ui/react';
+import { CloseIcon } from '@chakra-ui/icons';
+
+// TODO: Refactor states such that its a nested state
+const DonationsModal = ({ isOpen, onClose, data, setCurrentPageNum, loadInfo, isEdit }) => {
+  const { backend } = useBackend();
+  const toast = useToast();
+  const [categoryData, setCategoryData] = useState('Food');
+  const [typeData, setTypeData] = useState('');
+  const [weightData, setWeightData] = useState('lbs');
+  const [priceData, setPriceData] = useState('');
+  const [alertVisible, setAlertVisible] = useState(false);
+
+  useEffect(() => {
+    const getData = async () => {
+      try {
+        if (isOpen && isEdit) {
+          setCategoryData(data['category']);
+          setTypeData(data['item_name']);
+          setWeightData(data['quantity_type']);
+          setPriceData(data['price']);
+        } else if (isOpen && !isEdit) {
+          setCategoryData('Food');
+          setTypeData('');
+          setWeightData('lbs');
+          setPriceData('');
+        }
+      } catch (error) {
+        console.error('Error fetching data:', error);
+      }
+    };
+    getData();
+  }, [isOpen]);
```

Comment:
⚠️ **[eslint]** <[react-hooks/exhaustive-deps](https://github.com/facebook/react/issues/14920)> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>React Hook useEffect has missing dependencies: 'data' and 'isEdit'. Either include them or remove the dependency array. If 'setCategoryData' needs the current value of 'data', you can also switch to useReducer instead of useState and read 'data' in the reducer.